client/extensions/belgaCoverage/src/belga-coverage-editor.tsx

```diff
@@ -13,37 +13,39 @@ function getData(event: DragEvent) {
     return event.dataTransfer ? event.dataTransfer.getData(ALLOWED) : '';
 }
 
-export const BelgaCoverageEditor: React.StatelessComponent<IEditorComponentProps & {superdesk: ISuperdesk}> = (props) => {
-    if (props.value) {
-        return (
-            <div>
-                <BelgaCoverageInfo
-                    coverageId={props.value}
-                    removeCoverage={props.readOnly ? undefined : () => props.setValue(null)}
-                    superdesk={props.superdesk}
-                />
-                <BelgaCoverageImages
-                    coverageId={props.value}
-                    rendition={'thumbnail'}
-                    maxImages={3}
-                    superdesk={props.superdesk}
-                />
-            </div>
-        );
-    }
+export function getBelgaCoverageEditor(superdesk: ISuperdesk): React.StatelessComponent<IEditorComponentProps> {
+    return function BelgaCoverageEditor(props: IEditorComponentProps) {
```

Comment:
You can also use `return class A extends React.PureComponent {...}`.

Reply:
is there any benefit when doing it like that?

Reply:
That's the same benefits when using functional vs class components. Functional ones can't have refs or lifecycle methods. Pure component automatically implements `shoudComponentUpdate` (implements referential checking of props) and will perform better without much effort.

Reply:
isn't it more explicit then if it's not using any of those features?

Reply:
You can keep it if it's clearer for you. `React.Component` / `React.PureComponent` are basic building blocks so I don't see them as causing any issues.

Reply:
yep it's not causing any issues, I was just wondering if there is any typescript/etc benefits I'm not aware of